packages/test/src/test-nexus-handler.ts

```diff
@@ -0,0 +1,524 @@
+import { randomUUID } from 'node:crypto';
+import anyTest, { TestFn } from 'ava';
+import getPort from 'get-port';
+import * as nexus from 'nexus-rpc';
+import * as protoJsonSerializer from 'proto3-json-serializer';
+import * as temporalnexus from '@temporalio/nexus';
+import * as root from '@temporalio/proto';
+import * as testing from '@temporalio/testing';
+import { DefaultLogger, LogEntry, Runtime, Worker } from '@temporalio/worker';
+import { ApplicationFailure, CancelledFailure, defaultFailureConverter, defaultPayloadConverter, SdkComponent } from '@temporalio/common';
+import { cleanStackTrace } from './helpers';
+
+export interface Context {
+  httpPort: number;
+  taskQueue: string;
+  endpointId: string;
+  env: testing.TestWorkflowEnvironment;
+  logEntries: LogEntry[];
+}
+
+const test = anyTest as TestFn<Context>;
+
+test.before(async (t) => {
+  const logEntries = new Array<LogEntry>();
+
+  const logger = new DefaultLogger('INFO', (entry) => {
+    logEntries.push(entry);
+  });
+
+  Runtime.install({ logger });
+  t.context.httpPort = await getPort();
+  t.context.env = await testing.TestWorkflowEnvironment.createLocal({
+    server: { extraArgs: ['--http-port', `${t.context.httpPort}`] },
+  });
+  t.context.logEntries = logEntries;
+});
+
+test.after.always(async (t) => {
+  await t.context.env.teardown();
+});
+
+test.beforeEach(async (t) => {
+  const taskQueue = t.title + randomUUID();
+  const { env } = t.context;
+  const response = await env.connection.operatorService.createNexusEndpoint({
```

Comment:
I understand _this_ is the way for now, but what's the intent regarding this? Do we expect users to have to do the same in their own Temporal+Nexus tests?

Reply:
That's a good question. in Java we have a shortcut, we should probably make it easier in the test environment for Core based SDKs. @dandavison, wondering if you considered this in Python.

Reply:
No, I have not considered APIs for user Temporal Nexus testing yet; I'll add a TODO to the code. In my tests I have a helper function like this for creating Nexus endpoints.

Reply:
I think we need an issue to track this.

Reply:
👍 Created one for Python SDK-3837 "Python Nexus user testing utilities and docs"

Reply:
I would have made this a features issue with checkboxes for each SDK but 🤷, thanks!